### ISSUE TITLE
Output shape discrepancy between keras.applications.ResNet50 and original paper

### ISSUE THREAD

**Issue by daavoo** [keras-team/keras-applications#30]
In the tables of the original paper (and in other implementations [official tensorflow](https://github.com/tensorflow/models/blob/master/official/resnet/resnet_model.py#L518), [torchvision](https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py#L105), [keras-resnet](https://github.com/raghakot/keras-resnet/blob/master/resnet.py#L212), [tensorpack](https://github.com/tensorpack/tensorpack/blob/master/examples/ResNet/resnet_model.py#L120)) the output shape after the initial 3x3 max pooling is `56x56` but in this ResNet50 implementation the output shape is `55x55`.

I tried to add `padding=same` to [this line](https://github.com/keras-team/keras-applications/blob/master/keras_applications/resnet50.py#L233) but it looks like the provided imagenet weights were trained for the `55x55` output shape.

**Comment by taehoonlee** [comment]
Yes, you are right. That is because the three backends have different behaviors for `Conv2D(..., strides=2, padding='same', ...)`. I will investigate further and let you know. Thank you for the report @daavoo.

**Comment by taehoonlee** [comment]
@daavoo, Please see [the commit](https://github.com/keras-team/keras-applications/commit/36436f612e310fd5cc29dafd72538c664a6bdbd6). The original architecture has been reproduced and the accuracies have been slightly improved because of your help :) For more information, you can see [the report](https://github.com/taehoonlee/tensornets/issues/25) about the performance differences from different paddings.

**Comment by CMCDragonkai** [comment]
@daavoo What is the `pad_info` are you referring to?

**Comment by daavoo** [comment]
@CMCDragonkai It's from @taehoonlee 's tensornets. 
This function:
https://github.com/taehoonlee/tensornets/blob/master/tensornets/utils.py#L132
Example usage:
https://github.com/taehoonlee/tensornets/blob/master/tensornets/resnets.py#L69

**Comment by taehoonlee** [comment]
@CMCDragonkai, Please refer to [the report](https://github.com/taehoonlee/tensornets/issues/25) about different behaviors of `Conv2D(..., strides=2, padding='same')`. The `pad_info` is designed to handle the asymmetric padding of TensorFlow.
